gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -336,17 +337,34 @@ var gpii = fluid.registerNamespace("gpii");
         var promiseSequence = fluid.promise.sequence(promises);
         // snapshots is an array of snapshotted settingshandler results (one for each lifecycle action block)
         promiseSequence.then(function (snapshots) {
-            if (saveSnapshot) {
-                var toSnapshot = fluid.copy(solutionRecord);
-                toSnapshot.settingsHandlers = {};
-                fluid.each(snapshots, function (snapshot) {
-                    if (snapshot !== undefined) { // can be removed once everything is settingsHandlers (GPII-1235)
-                        fluid.each(snapshot, function (handlerBlock, blockKey) {
-                            toSnapshot.settingsHandlers[blockKey] = handlerBlock;
-                        });
-                    }
-                });
+            var toSnapshot = fluid.copy(solutionRecord);
+            toSnapshot.settingsHandlers = {};
+            fluid.each(snapshots, function (snapshot) {
+                if (snapshot !== undefined) { // can be removed once everything is settingsHandlers (GPII-1235)
+                    fluid.each(snapshot, function (handlerBlock, blockKey) {
+                        toSnapshot.settingsHandlers[blockKey] = handlerBlock;
+                    });
+                }
+            });
+            if (fullSnapshot) {
                 sessionState.originalSettings[solutionId] = toSnapshot;
+            } else {
+                // if we're doing an update, keep the settings that are already stored from the
+                // original snapshot, but augment it with any settings from the new snapshot
+                // that were not present in the original snapshot.
+                // First merge the basic structures:
+                var tmpOrigSettings = $.extend(true, {}, toSnapshot, sessionState.originalSettings[solutionId]);
```

Comment:
As it is, original values could actually be 'undefined' as well.. The reason I'm using $.extend here, is to avoid having to manually create the 'union' (or merged version) of the two objects. Instead I'm using extend for that, and then I fix the actual *setting* value copying in the below (in an 'undefined' friendly way)